src/App.jsx

```diff
@@ -14,33 +15,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const reset = () => {
+    setVisibleGoods(goodsFromServer);
+    setSortField('');
+    setIsReversed(false);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const sortByAlphabet = () => {
+    const sortedGoodsByAlphabet = [...visibleGoods].sort((good1, good2) => {
+      return isReversed
+        ? good2.localeCompare(good1)
+        : good1.localeCompare(good2);
+    });
+
+    setVisibleGoods(sortedGoodsByAlphabet);
+    setSortField(SORT_FIELD_ALPHABET);
+  };
+
+  const sortByLength = () => {
+    const sortedGoodsByLength = [...visibleGoods].sort((good1, good2) => {
+      return isReversed
+        ? good2.length - good1.length
+        : good1.length - good2.length;
+    });
+
+    setVisibleGoods(sortedGoodsByLength);
+    setSortField(SORT_FIELD_LENGTH);
+  };
+
+  const reverse = () => {
+    const reversedGoods = [...visibleGoods].reverse();
```

Comment:
You don't need to store goods in a state, try to simplify
Create a function that accepts goods array, sort type and reversed state and depending on these parameters prepare them. And move this function outside the function